crates/sui-core/src/consensus_handler.rs

```diff
@@ -114,6 +128,14 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
             Arc::new(consensus_output.sub_dag.leader.clone()),
         ));
 
+        // TODO: spawn a separate task for this as an optimization
+        update_low_scoring_authorities(
```

Comment:
My understanding is that on crash/recovery we have no guarantees of resuming the last known reputation scores. Shall this get addressed as part of this PR as well? Especially if the window we calculate the scores is not short-lived (ex 15 - 30 mins) , then simple restarts will make the validator see a different schedule compared to the others.

Reply:
I recall our discussion on the crash recovery here is that it is not needed since we will get the scores on the next subdag relatively quickly, and it is ok if the nodes are out of sync with the scores by design. Do you disagree with this?

Reply:
Also if a node is down and then comes back up, it spends some time in recovery mode, where is not making contributions by voting because it is fetching parent certificates etc. During recovery time, unless our recovery strategy gets to be much better than the time it takes to update a new score set, then the catching up node is going to be marked as low-scoring. The low scoring nodes do not apply the filtering logic, so it won't make a difference if it loads that info from disk to use right away, since it will be truly low scoring for at least a bit.